Vagrantfile

```diff
@@ -0,0 +1,78 @@
+  
+    # set up network ip and port forwarding
+    config.vm.network "forwarded_port", guest: 5000, host: 5000, host_ip: "127.0.0.1"
+    config.vm.network "private_network", ip: "192.168.33.10"
+  
+    # Windows users need to change the permission of files and directories
+    # so that nosetests runs without extra arguments.
+    # Mac users can comment this next line out
+    config.vm.synced_folder ".", "/vagrant", mount_options: ["dmode=775,fmode=664"]
+  
+    ######################################################################
+    # Create a virtual machine
+    ######################################################################
+    config.vm.provider "virtualbox" do |vb|
+      # Customize the amount of memory on the VM:
+      vb.memory = "512"
+      vb.cpus = 1
+      # Fixes some DNS issues on some networks
+      vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+      vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
+    end
+  
+    ######################################################################
+    # Copy some files to make developing easier
+    ######################################################################
+  
+    # Copy your .gitconfig file so that your git credentials are correct
+    if File.exists?(File.expand_path("~/.gitconfig"))
+      config.vm.provision "file", source: "~/.gitconfig", destination: "~/.gitconfig"
+    end
+  
+    # Copy your ssh keys for github so that your git credentials work
+    if File.exists?(File.expand_path("~/.ssh/id_rsa"))
+      config.vm.provision "file", source: "~/.ssh/id_rsa", destination: "~/.ssh/id_rsa"
+    end
+  
+    # Copy your ~/.vimrc file so that vi looks the same
+    if File.exists?(File.expand_path("~/.vimrc"))
+      config.vm.provision "file", source: "~/.vimrc", destination: "~/.vimrc"
+    end
+  
+    ############################################################
+    # Create a Python 3 environment for development work
+    ############################################################
+    config.vm.provision "shell", inline: <<-SHELL
+      # Update and install
+      apt-get update
+      apt-get install -y git tree wget python3-dev python3-pip python3-venv apt-transport-https
+      apt-get upgrade python3
+      # Create a Python3 Virtual Environment and Activate it in .profile
+      sudo -H -u vagrant sh -c 'python3 -m venv ~/venv'
+      sudo -H -u vagrant sh -c 'echo ". ~/venv/bin/activate" >> ~/.profile'
+      sudo -H -u vagrant sh -c '. ~/venv/bin/activate && cd /vagrant && pip install -r requirements.txt'
+    SHELL
+  
+    ######################################################################
+    # Add PostgreSQL docker container
+    ######################################################################
+    # docker run -d --name postgres -p 5432:5432 -v psql_data:/var/lib/postgresql/data postgres
+    config.vm.provision :docker do |d|
+      d.pull_images "postgres:alpine"
+      d.run "postgres:alpine",
```

Comment:
@YufeiGao6 Did you try running the vagrant environment and checking that postgres is actually coming up?

Reply:
I did vagrant up, everything seems fine to me. But I don't know how to check postgres.

Reply:
After doing ```vagrant ssh```, write the command ```docker images``` to see what docker images are present. To check logs of a a particular container, run ```docker container logs {containerName}```, for our case ```docker container logs postgres```. If the logs don't report any errors, we're good to go.

Reply:
I did this:

`YufeideMacBook-Pro:orders yufeigao$ postgres --help
postgres is the PostgreSQL server.
Usage:
  postgres [OPTION]...`

postgres should be coming up.

Reply:
After doing ```vagrant ssh``` run this command to get into the database: ```docker exec -ti postgres psql -U postgres```

Reply:
Yes, it worked. I'll merge this pull request.